tracer/src/Datadog.Trace/Ci/Agent/Payloads/EvpPayload.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="EvpPayload.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.Ci.Agent.Payloads
+{
+    /// <summary>
+    /// Event-platform payload
+    /// </summary>
+    internal abstract class EvpPayload
+    {
+        private Uri _url;
+
+        /// <summary>
+        /// Gets Event-Platform subdomain/track
+        /// </summary>
+        public abstract string EvpSubdomain { get; }
+
+        /// <summary>
+        /// Gets Event-Platform path
+        /// </summary>
+        public abstract string EvpPath { get; }
+
+        /// <summary>
+        /// Gets or sets the Payload url
+        /// </summary>
+        public Uri Url
+        {
+            get
+            {
+                if (_url is { } url)
+                {
+                    return url;
+                }
+
+                UriBuilder builder;
+                if (CIVisibility.Settings.Agentless)
+                {
+                    var agentlessUrl = CIVisibility.Settings.AgentlessUrl;
+                    if (!string.IsNullOrWhiteSpace(agentlessUrl))
+                    {
+                        builder = new UriBuilder(agentlessUrl);
+                        builder.Path = EvpPath;
+                    }
+                    else
+                    {
+                        builder = new UriBuilder(
+                            scheme: "https",
+                            host: $"{EvpSubdomain}.{CIVisibility.Settings.Site}",
+                            port: 443,
+                            pathValue: EvpPath);
+                    }
+                }
+                else
+                {
+                    // Use Agent EVP Proxy
+                    builder = new UriBuilder(CIVisibility.Settings.TracerSettings.Exporter.AgentUri);
+                    builder.Path = $"/evp_proxy/v1/{EvpPath}";
+                }
+
+                url = builder.Uri;
+                _url = url;
+                return url;
+            }
+
+            set
+            {
+                _url = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the payload contains events or not
+        /// </summary>
+        public abstract bool HasEvents { get; }
```

Comment:
nit: Is `HasEvents` ever _not_ defined as `bool HasEvents => Count > 0`? Could we just implement that here and not make it abstract?

Reply:
yes, the MultipartPayload have `Count > 1` because we have to add a dummy item in each payload (because... backend :shrug:)